### PR TITLE
Fix `GroupBy.value_counts` handling for multiple groupby columns

### LINKED ISSUE #6074: value_counts groupby multiple keys, wrong index
```python
import dask

ddf = dask.datasets.timeseries()
# simple column for value_counts
ddf['bool_tocount'] = False
ddf['bool_tocount'] = ddf['x'] < 0

pdf = ddf.head(10000)
```
expected result (pandas):
```python
pdf.groupby(['id', 'name'])['bool_tocount'].value_counts().head()
```

```bash
id   name    bool_tocount
884  Quinn   False           1
885  Bob     False           1
890  Oliver  True            1
891  Sarah   True            1
894  Dan     True            1
Name: bool_tocount, dtype: int64
```

wrong result (dask): wrong meaning index `bool_tocount` values are names and not True/False.
```python
ddf.groupby(['id', 'name'])['bool_tocount'].value_counts().head()
```

```bash
id   name    bool_tocount
877  Ray     Ray             2
880  Quinn   Quinn           2
882  Ingrid  Ingrid          3
883  Bob     Bob             2
     Oliver  Oliver          1
Name: bool_tocount, dtype: int64
```
I think it's the `level=1` in here: https://github.com/dask/dask/blob/ceec30358545599a6b978224cecb4e01b14dd517/dask/dataframe/groupby.py#L1858

Don't know the best way to fix it in a general manner. Ideas?

Thanks

### PR DESCRIPTION
**Pull request by charlesbluca**
- [x] Closes #6074
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: dask/dataframe/groupby.py [hunk 1]
```diff
@@ -2824,15 +2824,16 @@ def _unique_aggregate(series_gb, name=None):
...
 
 
 def _value_counts_aggregate(series_gb):
-    to_concat = {k: v.groupby(level=1).sum() for k, v in series_gb}
-    names = list(series_gb.obj.index.names)
-    return pd.Series(pd.concat(to_concat, names=names))
+    return pd.concat(
+        {k: v.groupby(level=-1).sum() for k, v in series_gb},
+        names=series_gb.obj.index.names,
+    )
```
Comment (charlesbluca) [inline dask/dataframe/groupby.py:2836]: It doesn't seem like we needed the `list` or `pd.Series` constructors here, so I've gone ahead and removed; interested to see if this causes any non-obvious failures

File: dask/dataframe/groupby.py [hunk 2]
```diff
@@ -2823,16 +2823,17 @@ def _unique_aggregate(series_gb, name=None):
 
 
 def _value_counts(x, **kwargs):
-    if len(x):
-        return M.value_counts(x, **kwargs)
-    else:
+    if not x.groups or all(pd.isna(k) for key in x.groups.keys() for k in key):
```
[the remark this answers is not in the document]
Comment (jrbourbeau) [reply dask/dataframe/groupby.py:2826]: This appears to be causing `dask/dataframe/tests/test_groupby.py::test_groupby_value_counts` to fail with 

```python
>   if not x.groups or all(pd.isna(k) for key in x.groups.keys() for k in key):
E   TypeError: 'int' object is not iterable

dask/dataframe/groupby.py:2826: TypeError
```

when grouping by a single column

```python
(Pdb) x.groups.keys()
dict_keys([1])
```

Would something this help?

```diff
diff --git a/dask/dataframe/groupby.py b/dask/dataframe/groupby.py
index 3b1e3b825..4cb96e0d7 100644
--- a/dask/dataframe/groupby.py
+++ b/dask/dataframe/groupby.py
@@ -10,6 +10,7 @@ import numpy as np
 import pandas as pd

 from dask import config
+from dask.core import flatten
 from dask.base import is_dask_collection, tokenize
 from dask.dataframe._compat import (
     PANDAS_GT_140,
@@ -2823,7 +2824,7 @@ def _unique_aggregate(series_gb, name=None):


 def _value_counts(x, **kwargs):
-    if not x.groups or all(pd.isna(k) for key in x.groups.keys() for k in key):
+    if not x.groups or all(pd.isna(key) for key in flatten(x.groups.keys())):
         return pd.Series(dtype=int)
     else:
         return x.value_counts(**kwargs)
(END)
```

[the remark this answers is not in the document]
Comment (charlesbluca) [reply dask/dataframe/groupby.py]: Yup, looks like that unblocks the failures 🎉

**Review by jrbourbeau — APPROVED** [review]
Thanks @charlesbluca!

**Pull request merged**